image_proc/src/crop_non_zero.cpp

```diff
@@ -78,8 +79,11 @@ CropNonZeroNode::CropNonZeroNode(const rclcpp::NodeOptions & options)
       }
     };
 
-  // Create publisher with same QoS as subscribed topic
-  auto qos_profile = getTopicQosProfile(this, "image_raw");
+  // Allow overriding QoS settings (history, depth, reliability)
+  pub_options.qos_overriding_options = rclcpp::QosOverridingOptions::with_default_policies();
+
+  // Create publisher with QoS matched to subscribed topic publisher
+  auto qos_profile = getTopicQosProfile(this, image_topic_);
```

Comment:
only change here is actually passing in image_topic_ rather than "image_raw" (technically still worked before, but leaves open the possibility of the topic names getting out of sync)